packages/calcite-components/src/components/action/action.scss

```diff
@@ -204,6 +204,13 @@
   }
 }
 
+:host([selection-appearance="highlight"]) {
```

Comment:
I think you can simplify this selector to `:host([selection-appearance="highlight"][active]) .button`